store/sqlstore/post_store.go

```diff
@@ -1735,6 +1735,39 @@ var specialSearchChar = []string{
 	":",
 }
 
+// GetRecentNthPostTime returns CreateAt time of most recent nth user-post
+func (s *SqlPostStore) GetRecentNthPostTime(n int64) (int64, error) {
+	if n <= 0 {
+		return 0, nil
+	}
+
+	builder := s.getQueryBuilder().
+		Select("CreateAt").
+		From("Posts p").
+		// Consider users posts only for cloud limit
+		Where(sq.And{
+			sq.Eq{"p.Type": ""},
+			sq.Expr("p.UserId NOT IN (SELECT UserId FROM Bots)"),
+		}).
+		OrderBy("p.CreateAt DESC").
+		Limit(1).
+		Offset(uint64(n - 1))
+
+	query, queryArgs, err := builder.ToSql()
+	if err != nil {
+		return 0, errors.Wrap(err, "GetRecentNthPostTime_tosql")
+	}
+
+	query = "SELECT COALESCE((" + query + "), 0)"
+
+	var createAt int64
+	if err := s.GetMasterX().Get(&createAt, query, queryArgs...); err != nil {
+		return 0, errors.Wrapf(err, "failed to get the Nth Post=%v", n)
```

Comment:
`n` is an integer, so the format specifier should be `%d`. Additionally, you need to pass the error as well.

Reply:
> format specifier should be %d

The %v already has the default format as %d for the int types. Is there any benefit of using %d explicitly?

> pass the error as well

I'm not sure if I follow correctly, but the first parameter is already the `err` field.

Reply:
> The %v already has the default format as %d for the int types. Is there any benefit of using %d explicitly?

Typically we use %v when we don't know the type or for errors. For integers, it's more idiomatic to use %d.

> I'm not sure if I follow correctly, but the first parameter is already the err field.

Ah my bad there. :+1: